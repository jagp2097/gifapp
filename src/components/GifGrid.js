import React, { useEffect, useState } from 'react';
import { Fragment } from 'react/cjs/react.production.min';
import GifGridItem from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  console.log(category)

  // const [images, setImages] = useState([]);
  const { data:images, loading } = useFetchGifs(category);

  // // si la informacion del componente cambia, este hook se ejecutarÃ¡
  // useEffect(() => {
  //   getGifs(category)
  //     .then(images => setImages(images));
  // }, [ category ]);

  return(
    <Fragment>
      <h3 className="animate__animated animate__fadeIn">{ category }</h3>

      { loading && <p>Cargando...</p> }

      <div className="card-grid">
        {
          images.map(image => 
            <GifGridItem 
              key={image.id}
              {...image}/>
            )
        }
      </div>
    </Fragment>
  );
};

export default GifGrid;
