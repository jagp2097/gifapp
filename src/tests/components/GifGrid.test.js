import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

// finge una llamada al archivo y controlar la info con la que va a responder
jest.mock('../../hooks/useFetchGifs'); 

describe('Pruebas al componente <GifGrid />', () => {

  const category = 'pepsi';

  test('Debe de mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={ category }/>);

    expect(wrapper).toMatchSnapshot();

  });

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'test'
    },
    {
      id: 'DES',
      url: 'https://localhost/cualquier/DSE.jpg',
      title: 'test2'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={ category }/>);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('p').exists()).toBe(false);

    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  });

});



