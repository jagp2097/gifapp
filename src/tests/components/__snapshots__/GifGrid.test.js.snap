// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    pepsi
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="DES"
      key="DES"
      title="test2"
      url="https://localhost/cualquier/DSE.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas al componente <GifGrid /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    pepsi
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
